### PR TITLE
check if there are differences in the ownerRefs, finalizers, and DeletionTimestamp in gc

### PR DESCRIPTION
**Pull request by cwdsuzhou**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

check if there are differences in the ownerRefs, finalizers, and DeletionTimestamp; if not, ignore the update.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: pkg/controller/garbagecollector/graph_builder.go [hunk 1]
```diff
@@ -134,14 +134,15 @@ func (gb *GraphBuilder) controllerFor(resource schema.GroupVersionResource, kind
...
 		UpdateFunc: func(oldObj, newObj interface{}) {
-			// TODO: check if there are differences in the ownerRefs,
-			// finalizers, and DeletionTimestamp; if not, ignore the update.
 			event := &event{
 				eventType: updateEvent,
 				obj:       newObj,
 				oldObj:    oldObj,
 				gvk:       kind,
 			}
+			if haveSameObjs(event) {
```
Comment (liggitt) [inline pkg/controller/garbagecollector/graph_builder.go]: what is the reason for short-circuiting this?

Comment (cwdsuzhou) [reply pkg/controller/garbagecollector/graph_builder.go]: Since the issue has been closed, could help review the PR

Comment (caesarxuchao) [reply pkg/controller/garbagecollector/graph_builder.go]: It's possible for the informer to deliver an updated event without delivering a create event first. In that case, the short circuiting here is not safe. The GC needs to process the update event at this [line](https://github.com/kubernetes/kubernetes/blob/32ec6c212ec9415f604ffc1f4c1f29b782968ff1/pkg/controller/garbagecollector/graph_builder.go#L560).

The performance improvement of this change is unclear. I would just remove the TODO.

**Comment by cwdsuzhou** [comment]
/close

reopen it if this PR can improve performance and catch `update` event

**Pull request closed without merging**